services/api-server/src/simcore_service_api_server/_service_function_jobs.py

```diff
@@ -317,3 +377,73 @@ async def map_function(
             user_id=self.user_id,
             product_name=self.product_name,
         )
+
+    async def function_job_outputs(
+        self,
+        *,
+        function: RegisteredFunction,
+        function_job: RegisteredFunctionJob,
+        user_id: UserID,
+        product_name: ProductName,
+        stored_job_outputs: FunctionOutputs | None,
+    ) -> FunctionOutputs:
+
+        from .api.routes import solvers_jobs_read, studies_jobs  # avoid circular import
```

Comment:
is there no way to restructure the code, by moving these parts in a module form which you can import them, in order to avoid this sort of runtime import?